loopd/swapclient_server.go

```diff
@@ -361,11 +361,20 @@ func (s *swapClientServer) GetLoopInQuote(ctx context.Context,
 
 	log.Infof("Loop in quote request received")
 
-	quote, err := s.impl.LoopInQuote(ctx, &loop.LoopInQuoteRequest{
+	htlcConfTarget, err := validateConfTarget(
+		req.ConfTarget, loop.DefaultHtlcConfTarget,
+	)
+	if err != nil {
```

Comment:
If `--external` is set, I think there's a reasonable expectation that the `htlc_conf` value is 0? If that's the case, `validateConfTarget` will just replace 0 with the default and not error anyway. 

New case here is if they set `--external` and an unacceptable conf target (eg 1), then the validation will fail even though external is true. 

What about something like this?
```
if external && contTarget!=0
  return err(external and conf target cannot both be set)
```
If both are being set, there's some missing understanding, so I think it's fair to error.

Reply:
Considering that rpc requests that go directly to loopd will have a zero conf target by default, I think it makes sense to remove the default values from loop (otherwise you're expecting a default value=6 from cli tool, and 0 from rpc requests). 

So now the server just checks that they aren't both set, which I think it the most simple. For non-external requests with 0 conf targets, it'll still overwrite them with the default. External will just have a 0 value written to disk + no validation because we fail early if the `htlc_conf` field is set.

Reply:
My idea was to check the `req.ExternalHtlc` _after_ the `validateConfTarget` and just see if it doesn't match the default value (which both rpc=0 and cli=6 defaults would be changed to).
That would have had the advantage that in the command line usage string, the `(default: 6)` would still have been shown.
But in `lnd` we usually set the command line default to `0` even if that's then set to another default value on the server side. So I'm fine with leaving it the way you now implemented it.